portfolio/src/main/webapp/script.js

```diff
@@ -102,7 +102,14 @@ function getComments() {
 class Countries {
 
   constructor(jsonObj) {
```

Comment:
It looks like we're going through each country and updating the keyedJSON map each time the user selects/un-selects a new checkbox. How about a slightly different data model, where you store the countries for each continent separately (perhaps in separate JSON objects)? The six boolean flags would then simply "toggle" which continents are "on" or "off" - without having to go through all the countries. 

Let me know what you think, and also feel free to postpone this to another PR since it could involve a large refactor.

Reply:
Ok I will try to refactor the code.

Reply:
Hi Vivek, I change the data model. It now has JSON objects for each continent. The new JSON objects will merge depending on the checkbox toggles.

Reply:
Thanks, this looks much cleaner!